### PR TITLE
Bump Fleet Server mTLS version gates to 8.19.19+, 9.3.8+, 9.4.4+

### LINKED ISSUE #9443: Enable mTLS transition E2E tests and version-gate Fleet Server mTLS support
## Summary

Once the upstream Elastic Agent bug [elastic/elastic-agent#13810](https://github.com/elastic/elastic-agent/issues/13810) is fixed, we need to:

1. Uncomment the mTLS transition test steps in `TestClientAuthRequired_FleetServerToAgent`
2. Consider version-gating Fleet Server mTLS support to require the fixed Elastic Agent version

## Background

PR [#9399](https://github.com/elastic/cloud-on-k8s/pull/9399) introduced mTLS support for Fleet Server to Elastic Agent connections. The E2E test `TestClientAuthRequired_FleetServerToAgent` was added to cover the auto-generated client certificate path, but the **transition test steps** (enabling/disabling mTLS) are currently commented out due to an upstream bug.

### Upstream Blocker

[elastic/elastic-agent#13810](https://github.com/elastic/elastic-agent/issues/13810) - Elastic Agent crashloops when transitioning from mTLS-enabled Fleet Server to disabled (and vice versa).

**Root cause:** The agent's stored configuration (`fleet.enc`) retains old client certificate paths. When mTLS is disabled and certificates are removed, the agent fails to start because it tries to load non-existent certificate files.

**Current workaround:** Users must manually clear the Elastic Agent state directory (`/usr/share/elastic-agent/state/`) when transitioning mTLS on or off.

### 1. Enable transition test steps

Once the upstream fix is released, uncomment the transition test code in `test/e2e/agent/client_auth_test.go`:

```go
// TODO: Uncomment transition tests once https://github.com/elastic/elastic-agent/issues/13810 is fixed.
// The upstream bug causes Elastic Agent to crashloop when transitioning from mTLS-enabled to disabled
// (and vice versa) because the agent's stored configuration retains old certificate paths.
//
// // Transition Fleet Server to client auth disabled.
// fsMutated := fleetServerBuilder.DeepCopy().WithMutatedFrom(&fleetServerBuilder)
// fsMutated.Agent.Spec.HTTP.TLS.Client.Authentication = false
// ...
```

### 2. Version-gate Fleet Server mTLS support

Add a minimum version check for Fleet Server mTLS to prevent users from enabling the feature on agent versions affected by the bug:

- Determine the first Elastic Agent version containing the fix
- Add validation in `pkg/controller/agent/` to emit a warning or block mTLS enablement on older versions
- Update documentation with minimum supported version

## Acceptance Criteria

- [ ] Transition test steps uncommented and passing
- [ ] Version gate implemented (warning or blocking)

## Related

- [elastic/elastic-agent#13810](https://github.com/elastic/elastic-agent/issues/13810) - Upstream bug (blocker)
- [#9399](https://github.com/elastic/cloud-on-k8s/pull/9399) - mTLS support for Fleet Server to Elastic Agent connections

### PR DESCRIPTION
**Pull request by pkoutsovasilis**
## Summary

Follow-up to #9486. After the original version gates (8.19.17+, 9.3.6+, 9.4.3+, 9.5.0+) were merged, a new stale-state bug in the Elastic Agent mTLS handshake was discovered: [elastic/elastic-agent#15408](https://github.com/elastic/elastic-agent/issues/15408). The fix has been backported to 8.19.19, 9.3.8, and 9.4.4; 9.5.0 remains the fix will be included by the time this is released.

This PR raises the minimum supported versions accordingly.

Closes https://github.com/elastic/cloud-on-k8s/issues/9443

## Changes

- Bump `FleetServerClientAuthSupported` thresholds: `8.19.17→8.19.19`, `9.3.6→9.3.8`, `9.4.3→9.4.4`

### CODE REVIEW

File: pkg/apis/agent/v1alpha1/validations_test.go [hunk 1]
```diff
@@ -960,10 +960,10 @@ func Test_checkClientAuthentication(t *testing.T) {
 			wantErr: true,
 		},
 		{
-			name: "authentication true, fleet server enabled, TLS enabled, version 9.4.3 (boundary): OK",
+			name: "authentication true, fleet server enabled, TLS enabled, version 9.4.4 (boundary): OK",
```
Comment (pebrc) [inline pkg/apis/agent/v1alpha1/validations_test.go:1008]: Could you add negative cases immediately below each new boundary (8.19.18, 9.3.7, and 9.4.3)? The existing negative versions are far below the thresholds, so they would pass even if the previous gates remained unchanged.

**Review by pebrc — COMMENTED** [review]
The version gate was raised to 8.19.19, 9.3.8, and 9.4.4, but the negative cases still use much older versions (8.15.0, 9.3.5, and 9.4.1). These tests would pass even if the old thresholds remained unchanged, so they do not protect the boundary changes introduced here. Please add rejection cases immediately below each new boundary: 8.19.18, 9.3.7, and 9.4.3, alongside the exact accepted boundary cases.

File: pkg/apis/agent/v1alpha1/validations_test.go (hunk 1 above)
Comment (pkoutsovasilis) [reply pkg/apis/agent/v1alpha1/validations_test.go:1008]: immediately below boundary version unit-tests added 8daa05dd2e35d823c5b6682e4e041a1733ec6878

**Review by moukoublen — APPROVED** [review]

**Review by pebrc — APPROVED** [review]

**Comment by pkoutsovasilis** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3.5|[](https://github.com/elastic/cloud-on-k8s/pull/9592)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)
